### PR TITLE
Replace Twig_SimpleFunction and Twig_Function with Timber_Function by extending both classes depending on Twig version

### PR DESCRIPTION
**Pull request by luism-s**
#### Issue

#1417

tl;dr: A fatal error was thrown by activating a WPML Translation Management plugin stating that `Twig_Function` was expected instead of `Twig_SimpleFunction`.

This only happens when timber is installed via composer in the theme folder. So the cause could be that the file that was extending `Twig_Function` into `Twig_SimpleFunction` (in Twig code) was loaded after all the plugins.

Another point is that Twig_Simple_Function [is getting deprecated in Twig 2.x](https://twig.sensiolabs.org/doc/1.x/deprecated.html), which Timber uses.

#### Solution
Search & Replace for `Twig_SimpleFunction` to `Twig_Function`.

#### Impact
Users should now be using `Twig_Function`, instead of `Twig_SimpleFunction`
Nothing that I'm aware of. The class named `Twig_SimpleFunction`that was extending `Twig_Function` was empty, so I guess there are no problems.

#### Usage
Use  `Twig_Function` instead

#### Considerations
To protect the ones still using `Twig_SimpleFunction`, we could create a file with the sole purpose of creating a class named  `Twig_SimpleFunction` and extending it from `Twig_Function`

# Update
Due to the discussion on #1478 I had the idea to instead of using `Twig_Function` or `Twig_SimpleFunction`, make a new one called `Timber_Function`, which extends from each one depending on the Twig version the site is running.

### CODE REVIEW

**Comment by luism-s** [comment]
oops, Travis failed, but only on PHP5, don't know why

**Comment by jarednova** [comment]
@luism-s thanks! I'm rerunning those PHP 5 tests, it said there was a MySQL error...

**Comment by luism-s** [comment]
@jarednova I made it. Only codecov is failing due to the large number of lines I changed on that file. I covered some lines of code with unit tests, but there's still some to go.

What do you think of this solution?

**Comment by jarednova** [comment]
@luism-s thanks for updating this! Glad to see it works (my confusion on #1478). My only question is whether the naming makes sense. Right now its full path is...

`Timber\Timber_Function`

... because the functionality is really about exposing functions to Twig, I'm wondering if `Timber\Twig_Function` might make more sense?

**Comment by luism-s** [comment]
Hm, OK, I really don't mind your name idea. I can do it later ;)

**Comment by luism-s** [comment]
@jarednova Renaming the function to Twig_Function made the tests in PHP5 fail. Maybe it's a conflict with Twig's Twig_Function? I don'know what I'm doing wrong, honestly

**Comment by luism-s** [comment]
Ok I figured out the problem. Good news, all tests checked 👌 ready to merge

**Comment by jarednova** [comment]
🎉  Hooray 🎉 ! Thanks so much @luism-s — wouldn't have thought that one would be so tough, but it really helps smooth out the Twig 2.0 integration. Thanks for all your work!!!

**Comment by luism-s** [comment]
All I wanted was really to be able to update Timber to the latest version, now I can :) When are you releasing the next release (pleonasm unintended)?

**Comment by jarednova** [comment]
@luism-s I like to do those on Tuesdays (my normal night for working late) — so look out for it tomorrow!
